### PR TITLE
JCLOUDS-664: Added DataVirtualHardDisk Representation With Tests

### PR DESCRIPTION
**Pull request by hsbhathiya**

### CODE REVIEW

**Comment by hsbhathiya** [comment]
@andreaturli  Can you review this?

File: azurecompute/src/main/java/org/jclouds/azurecompute/domain/DataVirtualHardDisk.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+   /**
+    * Specifies the Logical Unit Number (LUN) for the data disk. If the disk is the first disk that is added,
+    * this element is optional and the default value of 0 is used. If more than one disk is being added,
+    * this element is required.
+    * <p/>
+    * You can use Get Role to find the LUN numbers that are already being used.
+    * Valid LUN values are 0 through 31
+    */
+   @Nullable public abstract Integer LUN();
```
Comment (andreaturli) [inline azurecompute/src/main/java/org/jclouds/azurecompute/domain/DataVirtualHardDisk.java]: why upper-case?

File: azurecompute/src/main/java/org/jclouds/azurecompute/domain/DataVirtualHardDisk.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+   @Nullable public abstract URI mediaLink();
+
+   /**
+    * This property identifies the type of the storage account for the backing VHD.
+    * If the backing VHD is in an Provisioned Storage account, “Provisioned” is returned otherwise “Standard”
+    * is returned.
+    * <p/>
+    * This property is only returned with a version header of 2014-10-01 or newer
+    */
+   @Nullable public abstract String IOType();
```
Comment (andreaturli) [inline azurecompute/src/main/java/org/jclouds/azurecompute/domain/DataVirtualHardDisk.java]: why upper-case?

**Comment by hsbhathiya** [comment]
@andreaturli  I corrected those mistakes

**Comment by hsbhathiya** [comment]
@andreaturli  I have corected the pointed out mistakes. Shall we merge this?

File: azurecompute/src/main/java/org/jclouds/azurecompute/domain/DataVirtualHardDisk.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ * @see <a href="https://msdn.microsoft.com/en-us/library/azure/jj157193.aspx#DataVirtualHardDisks" >api</a>
+ */
+
+@AutoValue
+public abstract class DataVirtualHardDisk {
+
+   public enum Caching {
+      READ_ONLY,
+      READ_WRITE,
+      None
```
Comment (nacx) [inline azurecompute/src/main/java/org/jclouds/azurecompute/domain/DataVirtualHardDisk.java]: Is this case sensitive? Can this value be upper case too?

File: azurecompute/src/main/java/org/jclouds/azurecompute/xml/ListDataVirtualHardDisksHandler.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import org.jclouds.azurecompute.domain.DataVirtualHardDisk;
+import org.jclouds.http.functions.ParseSax;
+import org.xml.sax.Attributes;
+
+import java.util.List;
+
+public final class ListDataVirtualHardDisksHandler
+      extends ParseSax.HandlerForGeneratedRequestWithResult<List<DataVirtualHardDisk>> {
+   private boolean inDataVHD;
+   private final DataVirtualHardDiskHandler dataVirtualHardDiskHandler = new DataVirtualHardDiskHandler();
```
Comment (nacx) [inline azurecompute/src/main/java/org/jclouds/azurecompute/xml/ListDataVirtualHardDisksHandler.java]: Don't assign this here. Add a constructor that receives it as a parameter to get it injected instead.

**Comment by hsbhathiya** [comment]
@nacx I addressed those issues you mentioned. can we merge this?

File: azurecompute/src/main/java/org/jclouds/azurecompute/xml/ListDataVirtualHardDisksHandler.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.List;
+
+public final class ListDataVirtualHardDisksHandler
+      extends ParseSax.HandlerForGeneratedRequestWithResult<List<DataVirtualHardDisk>> {
+   private boolean inDataVHD;
+   private final DataVirtualHardDiskHandler dataVirtualHardDiskHandler;
+   private final Builder<DataVirtualHardDisk> VHDs = ImmutableList.builder();
+
+   @Inject
+   public ListDataVirtualHardDisksHandler(DataVirtualHardDiskHandler dataVirtualHardDiskHandler) {
```
Comment (nacx) [inline azurecompute/src/main/java/org/jclouds/azurecompute/xml/ListDataVirtualHardDisksHandler.java]: In general, injection constructors should be package protected. Just remove the public modifier:
```java
@Inject
ListDataVirtualHardDisksHandler(DataVirtualHardDiskHandler dataVirtualHardDiskHandler) {
```

**Comment by nacx** [comment]
@hsbhathiya One last minor comment. I'll merge it once addressed. Thanks!

**Comment by hsbhathiya** [comment]
@nacx  I made that change

**Comment by nacx** [comment]
Pushed to master as [80f5c8f](https://git-wip-us.apache.org/repos/asf?p=jclouds-labs.git;a=commit;h=80f5c8f763f731437ba68986b7d9837900544334). Thanks @hsbhathiya!

**Pull request closed without merging**